### PR TITLE
Add throwIfStringIsNull and throwIfObjectIsNull

### PR DESCRIPTION
**Pull request by p3dr0rv**
Add Utility functions to String and Objects 

used here https://github.com/AzureAD/ad-accounts-for-android/pull/1789

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java [hunk 1]
```diff
@@ -416,4 +416,13 @@ public static String base64Encode(final byte[] bytes) {
         }
         return sb.toString();
     }
+
+    /**
+     * A helper function for validating if the given String is null or empty.
+     */
+    public static void throwIfStringIsNull(@NonNull final String objectName, @Nullable final String object) throws IllegalStateException {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java]:
```suggestion
    public static void throwIfArgumentIsNullOrEmpty(@NonNull final String argumentName, @Nullable final String argument) throws IllegalStateException {
```

Comment (shahzaibj) [reply common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java]: Also you may want to throw a `NullPointerException` instead

File: common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java [hunk 2]
```diff
@@ -48,4 +49,13 @@ public static boolean equals(@Nullable final Object o1, @Nullable final Object o
         return (o1 == o2) || (o1 != null && o1.equals(o2));
     }
 
+    /**
+     * A helper function for validating if the given object is null or empty.
+     */
+    public static void throwIfObjectIsNull(@NonNull final String objectName, @Nullable final Object object) throws IllegalStateException {
+        if (object == null) {
+            throw new IllegalStateException(objectName + " is null or empty.");
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]: You may want to throw a `NullPointerException` instead

Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]:
```suggestion
            throw new IllegalStateException(objectName + " is null.");
```

File: common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java [hunk 3]
```diff
@@ -48,4 +49,13 @@ public static boolean equals(@Nullable final Object o1, @Nullable final Object o
         return (o1 == o2) || (o1 != null && o1.equals(o2));
     }
 
+    /**
+     * A helper function for validating if the given object is null or empty.
+     */
+    public static void throwIfObjectIsNull(@NonNull final String objectName, @Nullable final Object object) throws IllegalStateException {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]:
```suggestion
    public static void throwIfArgumentIsNull(@NonNull final String argumentName, @Nullable final Object argument) throws IllegalStateException {
```

File: common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java (hunk 1 above)
Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java]: 7cee1dd4643a39521911892cd529e5ca2bfae6fb

File: common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java (hunk 2 above)
Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]: 7cee1dd4643a39521911892cd529e5ca2bfae6fb

File: common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java (hunk 3 above)
Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]: 7cee1dd4643a39521911892cd529e5ca2bfae6fb

File: common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java (hunk 2 above)
Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/util/ported/ObjectUtils.java]: 7cee1dd4643a39521911892cd529e5ca2bfae6fb

File: common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java (hunk 1 above)
Comment (p3dr0rv) [reply common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java]: Thanks for the feedback @shahzaibj, great suggestions

**Review by shahzaibj — APPROVED** [review]
LGTM! Will be good to add link to some other PR(s) where these methods are going to be used.

**Review by rpdome — APPROVED** [review]

**Pull request merged**